### PR TITLE
Jetpack Pricing Cart Integration - Handle CTA Label and Cart change 

### PR DESCRIPTION
**Pull request by vikneshwar**
#### Proposed Changes

- Changes to show the button label as **View Cart** with icon when an item is added to cart
- Clicking of **View Cart** should navigate to checkout page
- Add checks for CRM products and Bundles - which should remain as it was before
- Wrap the Pricing page component with `CartSubscriber` which listens to cart and forces re-render when cart changes

#### Testing Instructions

* click on Jetpack Cloud live link below
    * Goto `/pricing`
* or boot up this PR 
    * Run `git fetch && git checkout fix/fix/view-cart-cta`
    * Run `yarn start-jetpack-cloud`
    * Goto http://jetpack.cloud.localhost:3000/pricing
* Switch to Bundle section and confirm the button Label on the card is the same as before and confirm that clicking on it takes you to checkout page
* Switch to the Product section and confirm the following
    * All the product cards (except the one already purchased and CRM) CTA label should be  **Add to Cart** instead of **Get**
    * Clicking on the CTA adds the item to the cart in header and button label for the item changes to **View Cart**
    * Make sure the **View Cart** CTA matches the design
    * Click on **View Cart** should redirect to checkout page
    * Go back to pricing page and remove an item from cart and confirm that the CTA for that item changes back to **Add to Cart**
    * Confirm that for CRM CTA says **Get** and clicking on it takes to CRM page
  
 
#### Video

[Desktop-1670928897817.webm](https://user-images.githubusercontent.com/2027003/207299972-1fbfd911-dc50-445d-8c68-15b6e404bef0.webm)

#### Products page
| Before | After |
|--------|------|
| Screen Shot 2022-12-06 at 3 26 40 PM | Screen Shot 2022-12-06 at 4 33 54 PM |

#### Bundles page
| Before | After |
|--------|------|
| Screen Shot 2022-12-06 at 3 26 40 PM | Screen Shot 2022-12-06 at 4 33 54 PM |

#### Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to #
1203495437860512-as-1203534015891085/f
1203495437860512-as-1203534015891165/f

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203534015891085
  - https://app.asana.com/0/0/1203534015891165

### CODE REVIEW

**Review by andrii-lysenko — APPROVED** [review]
Thank you for these changes. Looks good, we could merge it into main branch and continue working on it.

File: client/my-sites/plans/jetpack-plans/product-store/hooks/use-store-item-info.tsx [hunk 1]
```diff
@@ -157,24 +155,62 @@ export const useStoreItemInfo = ( {
...
+
 	const getCheckoutURL = useCallback(
 		( item: SelectorProduct ) => {
-			return shouldShowCart
-				? '#'
-				: createCheckoutURL?.( item, getIsUpgradeableToYearly( item ), getPurchase( item ) );
+			if ( getShouldShowCart( item ) ) {
+				if ( getIsProductInCart( item ) ) {
+					//open the cart
+					return `http://www.wordpress.com/checkout/${ siteSlug }`;
```
Comment (sirbrillig) [inline client/my-sites/plans/jetpack-plans/product-store/hooks/use-store-item-info.tsx:175]: Shouldn't this be `https`?

Also, what if the user is on a different host like `wpcalypso` or `horizon`? You might want to reconstruct the URL from `window.location`, maybe [like we do here](https://github.com/Automattic/wp-calypso/blob/f9f6c47cb4bb3770ce428516bbb32646017ef4f1/client/my-sites/checkout/composite-checkout/lib/pending-page.ts#L165-L176).

File: client/my-sites/plans/jetpack-plans/cart-subscriber.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import { cartManagerClient } from 'calypso/my-sites/checkout/cart-manager-client';
+export type CartSubscriberProps = {
+	siteId: number | null;
+	render: () => JSX.Element;
+};
+export default function ( { siteId, render }: CartSubscriberProps ) {
+	//simulate forceUpdate to manually trigger re-render on cart changes
+	/* eslint-disable  @typescript-eslint/no-unused-vars */
+	const [ ignored, forceUpdate ] = useReducer( ( x ) => x + 1, 0 );
+	useEffect( () => {
```
Comment (sirbrillig) [inline client/my-sites/plans/jetpack-plans/cart-subscriber.tsx:11]: This is an interesting component! How is this different from `useShoppingCart` and `CalypsoShoppingCartProvider`? Any component that calls `useShoppingCart` will already be subscribed to cart changes and will re-render when they occur.

File: client/my-sites/plans/jetpack-plans/cart-subscriber.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { useEffect, useReducer } from 'react';
+import { cartManagerClient } from 'calypso/my-sites/checkout/cart-manager-client';
+export type CartSubscriberProps = {
+	siteId: number | null;
+	render: () => JSX.Element;
+};
+export default function ( { siteId, render }: CartSubscriberProps ) {
+	//simulate forceUpdate to manually trigger re-render on cart changes
+	/* eslint-disable  @typescript-eslint/no-unused-vars */
+	const [ ignored, forceUpdate ] = useReducer( ( x ) => x + 1, 0 );
```
Comment (sirbrillig) [inline client/my-sites/plans/jetpack-plans/cart-subscriber.tsx:10]: You can remove this eslint disable line by just removing `ignored`.

```
	const [ , forceUpdate ] = useReducer( ( x ) => x + 1, 0 );
```

File: client/my-sites/plans/jetpack-plans/product-store/hooks/use-store-item-info.tsx [hunk 4]
```diff
@@ -57,30 +58,27 @@ export const useStoreItemInfo = ( {
 	const isMultisite = useSelector(
 		( state ) => !! ( siteId && isJetpackSiteMultiSite( state, siteId ) )
 	);
-
+	const siteSlug = useSelector( ( state ) => getSiteSlug( state, siteId ) ) as string;
```
Comment (sirbrillig) [inline client/my-sites/plans/jetpack-plans/product-store/hooks/use-store-item-info.tsx:61]: What happens if `siteSlug` is null? By using `as string`, you're removing some of the type safety in this code.

It looks to me like the result is that `getCheckoutURL()` will return `http://www.wordpress.com/checkout/null` if that happens, which is probably not where you want to send the user.

**Review by sirbrillig — COMMENTED** [review]
I know this is part of a work-in-progress, but I wanted to leave a few drive-by comments anyway. If they're not helpful, feel free to ignore them!

File: client/my-sites/plans/jetpack-plans/cart-subscriber.tsx (hunk 2 above)
Comment (andrii-lysenko) [reply client/my-sites/plans/jetpack-plans/cart-subscriber.tsx:11]: Thanks for the suggestion. When initially developed we had an issue when with `useShoppingCart` there was an update when we add something there, however, after removing the state wasn't updated automatically. We would definitely give it another look.

**Comment by vikneshwar** [comment]
@sirbrillig, thanks much for your valuable feedback, we will have these and a few more refactoring addressed in our upcoming PRs.

**Pull request merged**